src/main/java/com/commercetools/emailprocessor/email/EmailProcessor.java

```diff
@@ -80,8 +82,9 @@ public static EmailProcessor of() {
 
         // 1. Query for all custom objects and map each to the future chain above.
         // 2. Map the list of future to an allOf future to execute them in parallel.
+        // TODO: Need to pass correlation id here to queryAll
         return queryAll(client, query, customObjectMapper)
-                .thenCompose(stages -> allOf(stages.toArray(new CompletableFuture[stages.size()])))
+                .thenCompose(stages -> allOf(stages.toArray(new CompletableFuture[0])))
```

Comment:
why did you replace it with 0?

Reply:
"In older Java versions using pre-sized array was recommended, as the reflection call which is necessary to create an array of proper size was quite slow. However since late updates of OpenJDK 6 this call was intrinsified, making the performance of the empty array version the same and sometimes even better, compared to the pre-sized version. Also passing pre-sized array is dangerous for a concurrent or synchronized collection as a data race is possible between the size and toArray call which may result in extra nulls at the end of the array, if the collection was concurrently shrunk during the operation.
This inspection allows to follow the uniform style: either using an empty array (which is recommended in modern Java) or using a pre-sized array (which might be faster in older Java versions or non-HotSpot based JVMs)."